services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -87,10 +87,20 @@ def get_delete_operations(all_repo_files: set[str], split_names: set[str], confi
     existing_split_pattern = re.compile(
         f"^({'|'.join(re.escape(f'{config}/{split_name}') for split_name in split_names)})/"
     )
+    # For directories like "partial-train" for the file at "en/partial-train/0000.parquet" in the C4 dataset.
+    # Note that "-" is forbidden for split names so it doesn't create directory names collisions.
+    # caveat: the split could become full processed
+    existing_partial_split_pattern = re.compile(
+        f"^({'|'.join(re.escape(f'{config}/partial-{split_name}') for split_name in split_names)})/"
+    )
+
     return [
         CommitOperationDelete(path_in_repo=file)
         for file in all_repo_files
-        if same_config_pattern.match(file) and not existing_split_pattern.match(file)
+        if same_config_pattern.match(file)
+        and file.endswith(".duckdb")
```

Comment:
indeed: it seems safe that the duckdb job only deletes duckdb files, and not parquet files